src/mapped-types.ts

```diff
@@ -412,12 +412,18 @@ export type PromiseType<T extends Promise<any>> = T extends Promise<infer U>
  *   };
  *   type ReadonlyNestedProps = DeepReadonly<NestedProps>;
  */
-export type DeepReadonly<T> = T extends (...args: any[]) => any
+export type DeepReadonly<T> = T extends
+  | ((...args: any[]) => any)
+  | string
+  | number
+  | boolean
```

Comment:
I would replace it with Primitive type, it will guard against other edge cases

Reply:
Didn't realize there was one, perfect.